rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java

```diff
@@ -251,6 +259,9 @@ public void rollback() throws SQLException {
 
     @Override
     public void setAutoCommit(boolean autoCommit) throws SQLException {
+        if (exception != null) {
+            throw exception;
```

Comment:
why throw exx ? the connection will  remain unavailable?

Reply:
execution should not continue because an exception has already been made

Reply:
agree with @slievrly